core/src/main/scala/org/bitcoins/core/script/constant/Constants.scala

```diff
@@ -191,6 +191,15 @@ sealed abstract class ScriptNumberOperation
     extends ScriptNumber
     with ScriptOperation {
   override def hex = opCode.toHexString
+
+  override def equals(obj: Any): Boolean = {
+    obj match {
+      case number: ScriptNumber =>
+        number.toLong == toLong
+      case _ =>
+        super.equals(obj)
+    }
+  }
```

Comment:
For the record, this was a bug where we had `OP_TRUE == OP_1` and `OP_FALSE == OP_0` as `false`